spec/controllers/rentals_controller_spec.rb

```diff
@@ -1,5 +1,148 @@
 require 'rails_helper'
 
 RSpec.describe RentalsController, type: :controller do
+  include Devise::TestHelpers
+
+  describe 'GET #new' do
+    context "current_client signed in" do
+      before(:each) do
+        @client = FactoryGirl.create(:client)
+        sign_in @client
+      end
+      it "assigns a new Rental to @rental" do
+        get :new
+        expect(assigns(:rental)).to be_a_new(Rental) 
+      end
+      it "renders the :new template" do
+        get :new
+        expect(response).to render_template :new
+      end
+    end
+    context "current_user signed in" do
+      before(:each) do
+        @user = FactoryGirl.create(:user)
+        sign_in @user
+      end
+      it "redirects user to root" do
+        get :new
+        expect(response).to redirect_to "/"
+      end
+    end
+  end
+
+  describe 'POST #create' do
+    before(:each) do
+      @client = FactoryGirl.create(:client)
+      @user = FactoryGirl.create(:user)
+    end
+    context "current_client signed in" do
+      before(:each) do
+        sign_in @client
+      end
+      it "assigns an id equal to the currently logged in CLIENT's id" do
+        post :create
+        expect(assigns(:id)).to eq @client.id
+      end
+    end
+    context "with valid attributes" do
+      before(:each) do
+        sign_in @client
+      end
+      it "saves the new rental in the database" do
+        post :create, FactoryGirl.attributes_for(:rental)
+        rentals = Rental.all
+        expect(rentals.length).to eq(1)
```

Comment:
it's usually better to assert the change in total instead of saying the count you expect.
```ruby
expect {
  post :create, FactoryGirl.attributes_for(:rental)
}.to change(Rental,:count).by(1)
```
I think I didn't have this the right way in mine when I was copied because it's the good way down below for the delete.